include/libtorrent/aux_/session_impl.hpp

```diff
@@ -558,7 +585,17 @@ namespace libtorrent
 			void set_peer_id(peer_id const& id);
 			void set_key(std::uint32_t key);
 			std::uint16_t listen_port() const override;
+			std::uint16_t listen_port(listen_socket_t* sock) const;
 			std::uint16_t ssl_listen_port() const override;
+			std::uint16_t ssl_listen_port(listen_socket_t* sock) const;
+
+			void for_each_listen_socket(std::function<void(aux::session_listen_socket*)> f) override
```

Comment:
I used a pointer here because other places in the code use such pointers as handles. I could pass a reference here but then the function would just end up converting it to a pointer.